core/node/libp2p/rcmgr.go

```diff
@@ -236,86 +219,45 @@ func NetLimit(mgr network.ResourceManager, scope string) (rcmgr.BasicLimitConfig
-		})
-		return result, err
-
+		return result, mgr.ViewTransient(func(s network.ResourceScope) error { return getLimit(s) })
 	case strings.HasPrefix(scope, config.ResourceMgrServiceScopePrefix):
 		svc := strings.TrimPrefix(scope, config.ResourceMgrServiceScopePrefix)
-		err := mgr.ViewService(svc, func(s network.ServiceScope) error {
-			return getLimit(s)
-		})
-		return result, err
-
+		return result, mgr.ViewService(svc, func(s network.ServiceScope) error { return getLimit(s) })
 	case strings.HasPrefix(scope, config.ResourceMgrProtocolScopePrefix):
 		proto := strings.TrimPrefix(scope, config.ResourceMgrProtocolScopePrefix)
-		err := mgr.ViewProtocol(protocol.ID(proto), func(s network.ProtocolScope) error {
-			return getLimit(s)
-		})
-		return result, err
-
+		return result, mgr.ViewProtocol(protocol.ID(proto), func(s network.ProtocolScope) error { return getLimit(s) })
 	case strings.HasPrefix(scope, config.ResourceMgrPeerScopePrefix):
 		p := strings.TrimPrefix(scope, config.ResourceMgrPeerScopePrefix)
 		pid, err := peer.Decode(p)
 		if err != nil {
 			return result, fmt.Errorf("invalid peer ID: %q: %w", p, err)
 		}
-		err = mgr.ViewPeer(pid, func(s network.PeerScope) error {
-			return getLimit(s)
-		})
-		return result, err
-
+		return result, mgr.ViewPeer(pid, func(s network.PeerScope) error { return getLimit(s) })
 	default:
 		return result, fmt.Errorf("invalid scope %q", scope)
 	}
 }
 
 // NetSetLimit sets new ResourceManager limits for the given scope. The limits take effect immediately, and are also persisted to the repo config.
-func NetSetLimit(mgr network.ResourceManager, repo repo.Repo, scope string, limit rcmgr.BasicLimitConfig) error {
+func NetSetLimit(mgr network.ResourceManager, repo repo.Repo, scope string, limit rcmgr.BaseLimit) error {
 	setLimit := func(s network.ResourceScope) error {
 		limiter, ok := s.(rcmgr.ResourceScopeLimiter)
 		if !ok { // NullResourceManager
 			return NoResourceMgrError
 		}
 
 		var newLimit rcmgr.Limit
-		if limit.Dynamic {
-			newLimit = &rcmgr.DynamicLimit{
-				MemoryLimit: rcmgr.MemoryLimit{
-					MemoryFraction: limit.MemoryFraction,
-					MinMemory:      limit.MinMemory,
-					MaxMemory:      limit.MaxMemory,
-				},
-				BaseLimit: rcmgr.BaseLimit{
-					Streams:         limit.Streams,
-					StreamsInbound:  limit.StreamsInbound,
-					StreamsOutbound: limit.StreamsOutbound,
-					Conns:           limit.Conns,
-					ConnsInbound:    limit.ConnsInbound,
-					ConnsOutbound:   limit.ConnsOutbound,
-					FD:              limit.FD,
-				},
-			}
-		} else {
-			newLimit = &rcmgr.StaticLimit{
-				Memory: limit.Memory,
-				BaseLimit: rcmgr.BaseLimit{
-					Streams:         limit.Streams,
-					StreamsInbound:  limit.StreamsInbound,
-					StreamsOutbound: limit.StreamsOutbound,
-					Conns:           limit.Conns,
-					ConnsInbound:    limit.ConnsInbound,
-					ConnsOutbound:   limit.ConnsOutbound,
-					FD:              limit.FD,
-				},
-			}
+		newLimit = &rcmgr.BaseLimit{
```

Comment:
Why not use a copy? I’m afraid we’ll forget to manually copy something in here in the future.

Reply:
There's no point in introducing a `newLimit` in the first place, now that the dynamic limit is gone.